docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,200 @@
+| Module uses AmazonS3Client API's to upload objects into S3. Large files will upload using multipart feature and small files(Number of blocks is 1) will upload using putObject(...) API. | Operator uses Hadoop filesystems like S3AFileSystem, S3NFileSystem, etc. Consists of couple of steps to upload object into S3: (1) Write the data into the local filesystem. (2) When the stream close, filesystem uploads the local object into S3. |
+| If a block fails to upload, then you need to re-upload the data for that block only.                                                                                                     | If a file fails to upload then you need to re-upload the complete file.                                                                                                                                                                              |
+
+### About Multipart Upload Feature
+----------------------------------
+Uploading the blocks of a file into S3 can be achieved by [multipart feature](http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html).
+
+Using Multipart Upload feature, we can upload a single large file in multiple smaller parts. In Multipart Upload process, each part is a portion of object's data. Each part can be uploaded independently. 
+After all parts of a file are uploaded successfully, Amazon S3 represents the data as a single object.
+
+Please refer the [Java code](http://docs.aws.amazon.com/AmazonS3/latest/dev/llJavaUploadFile.html) for uploading file into Amazon S3 bucket using multipart feature.
+
+### About S3 Output Module
+--------------------------
+
+Purpose of S3Output module is to upload files/directories into an Amazon S3 bucket using the multipart upload feature.
+
+S3Output module is **fault-tolerant**, **statically/dynamically partitionable** and has **exactly once** semantics.
+
+Module class is **S3OutputModule** located in the package **org.apache.apex.malhar.lib.fs.s3**; please refer to [github URL](https://github.com/apache/apex-malhar/blob/master/library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3OutputModule.java).
+
+### About the operators in S3OutputModule
+-----------------------------------------
+
+Following diagram illustrates the DAG in this module:
+
+![](images/s3output/s3outputmodule.png)
+
+-   ***S3InitateFileUploadOperator***
+    -   Initiate the upload for the file using AmazonS3Client.initiateMultipartUpload(...) method only if the number of blocks for a file is greater than 1. By Successfully initiating the file, S3 returns a response of type InitiateMultipartUploadResult, which includes the upload ID, which is the unique identifier for the multipart upload. This upload ID must include whenever upload the blocks, complete an upload of a file.
```

Comment:
Highlight literal names (i.e. variable, class, function, etc.) in some way (e.g. backquotes).